### DISCUSSION TITLE
Older Android tablet stopped syncing ical feeds, with error "Certificate Chain not trusted."

### DISCUSSION THREAD

**Discussion by jasonsilver** [bitfireAT/icsx5#466] [Q&A] [answered]
This is just a problem on the older tablet (Android 9.0, kernel 3.18.35), no issue on my newer Android 14 Pixel 8 phone - wondering if it's actually an Android OS issue?

**Comment by sunkup** [comment]
Sounds like it. You might want to check whether you are using user certificates on the tablet.

**Reply by UserX404** [reply]
I have the same issue with [Mozilla's Holiday Calendar](ttps://www.mozilla.org/media/caldata/AustrianHolidys.ics) on my Galaxy Tab 2 with LOS 13 (Android 6.0.1)
There is no entry in trusted user certs. Just system certs are listed.

I imported my calendars from my nextcloud and davx5 and have no such issue on all my other devices.
Any idea how to solve/circumvent this issue?

**Reply by sunkup** [reply]
Just tried with an Android 6 emulator and it's working fine here. 

I am not completely sure, but without recent updates older Android devices might not have the latest root certificates. You might want to look into manually installing required (root) certificates. Never did it myself though, so can't help with that.

**Comment by jimboolio** [comment, marked as answer]
I encountered the same issue

My experience:
1. The prompt asked whether I trust the certificate SHA1 and SHA256 of domain uef.fi and [*].uef.fi
2. I checked the certificate fingerprint from peppi.uef.fi since I assumed that was the calendar source (it wasn't)
3. There was huge list of covered subdomains, and the fingerprint did not match -> I did not accept the certificate
4. I managed to close the prompt
5. After closing the prompt icsx5 showed that the cert error was with domain elearn.uef.fi which uses different cert, but I could not re-trigger the prompt, since the cert in the previous prompt was valid, but I did not realize it
6. I managed to delete the calendar from icsx5 but could not re-add it since the cert was untrusted and it did not re-trigger the prompt, I basically got stuck here for some time
7. I discovered that you can work-around the issue if you have davx5 installed, by opening the davx5 settings and clicking the `reset (un)trusted certificates button`
8. Now returning to icsx5, tried to re-add the calendar, and sure thing the certificate prompt re-appears and the certificate was indeed correct one but elearn.uef.fi domain has different cert than peppi.uef.fi

From usability perspective:
- At step 3, it would be very useful to know which subdomain is the certificate validation prompt about
- At step 5 there could be some further info about how the issue could be resolved
- Step 7 is in my opinion possible workaround, since I assume the issue is likely quite rare. If the usability isn't otherwise improved, the resolution workaround could be mentioned on the website. Or if the issue is too rare for even that at least there is this comment. Hoping this helps someone!

Thanks for the great app!

**Reply by jasonsilver** [reply]
Thanks for the great insights! I appreciate your comment. I've abandoned the problem because I decided to use SyncThing to syncronize the ical file after downloading from my Desktop where the certificate issue isn't a problem. 
I'll mark your solution as the answer. Thanks again for your thoughtful interaction with this post.
